ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsontop/v1/tabsontop/clientlibs/site/js/tabs.js

```diff
@@ -148,32 +132,31 @@
             return this.element.querySelector(Tabs.selectors.qm);
         }
 
+        #getTabListElement() {
+            return this.element.querySelector(Tabs.selectors.olTabList)
+        }
+
         #syncTabLabels(){
-            var tabs = this._elements["tab"];
-            if (tabs) {
-                if(Array.isArray(tabs)){
-                    for (var i = 0; i < tabs.length; i++) {
-                        tabs[i].id = this._elements["tabPanelElementId"][i] + "__tab";
-                        tabs[i].setAttribute("aria-controls", this._elements["tabPanelElementId"][i] + "__tabpanel");
-                    }
-                }else{
-                    tabs.id = this._elements["tabPanelElementId"][0] + "__tab";
-                    tabs.setAttribute("aria-controls", this._elements["tabPanelElementId"][0] + "__tabpanel");
+            var tabs = this.#getCachedTabs();
+            var tabPanels = this.#getCachedTabPanels();
+            if (tabPanels) {
+                for (var i = 0; i < tabPanels.length; i++) {
+                    var childViewId = tabPanels[i].querySelectorAll(
+                        '[data-cmp-adaptiveformcontainer-path="' + this.options['adaptiveformcontainerPath'] + '"]')[0].id;
+                    tabs[i].id = childViewId + "__tab";
+                    tabs[i].setAttribute("aria-controls", childViewId + "__tabpanel");
                 }
             }
         }
 
         #syncTabPanels(){
-            var tabPanels = this._elements["tabpanel"];
+            var tabPanels = this.#getCachedTabPanels();
             if(tabPanels) {
-                if(Array.isArray(tabPanels)) {
-                    for (var i = 0; i < tabPanels.length; i++) {
-                        tabPanels[i].id = this._elements["tabPanelElementId"][i] + "__tabpanel";
-                        tabPanels[i].setAttribute("aria-labelledby", this._elements["tabPanelElementId"][i] + "__tab");
-                    }
-                }else{
-                    tabPanels.id = this._elements["tabPanelElementId"][0] + "__tabpanel";
-                    tabPanels.setAttribute("aria-labelledby", this._elements["tabPanelElementId"][0] + "__tab");
+                for (var i = 0; i < tabPanels.length; i++) {
+                        var childViewId = tabPanels[i].querySelectorAll(
+                            '[data-cmp-adaptiveformcontainer-path="' + this.options['adaptiveformcontainerPath'] + '"]')[0].id;
```

Comment:
this can result in NPE. 
pls check for null before resolving the id.

Reply:
@nitin, currently tab panel can't exist without a child element, if that would be the case then sightly itself will break.